src/serlo-editor/core/editor.tsx

```diff
@@ -132,10 +190,18 @@ function InnerDocument({
   if (!isInitialized) return null
 
   return (
-    <div className="relative" ref={wrapperRef}>
+    <div
+      className="relative"
+      ref={(div) => {
+        wrapperRef.current = div
+        div?.addEventListener('focusin', onFocusinHandler)
+      }}
```

Comment:
```suggestion
      ref={wrapperRef}
```

better to handle attaching the eventListener in a useEffect.

```
useEffect(() => {
  if (wrapperRef.current) {
      wrapperRef.current.addEventListener('focusin', onFocusinHandler);

      return () => {
          wrapperRef.current.removeEventListener('focusin', onFocusinHandler);
      };
  }
}, [onFocusinHandler]);
```

Reply:
This was my first attempt but it failed for me. In the first rendering wrapperRef.current is null and since it is a reference the component does not get rendered again once the HTML Element is mounted (that's my theory - the bug was that the listener was not added property). Maybe I did something else wrong. I would be happy if someone can fix this.

Reply:
Fixed.

- Conditional rendering of the component means that the component renders `null`, until `initRootSaga` and its recursion is complete. Once this is done, the `isInitialized` state value is switched to `true`. Therefore, we can use `isInitialized` as a dependency for the event attaching `useEffect`.
- Compared to Mikey's example, `wrapperRef.current` was moved into a constant, because `wrapperRef.current` usually no longer exists when the clean-up function is called, so the event listener doesn't get removed.
- The condition for setting the focus path was moved inside the setter, so that the `focusPath` wouldn't be a dependency for `onFocusinHandler` callback. This way, we avoid `onFocusinHandler` being re-declared whenever `focusPath` changes.